pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/leap/LeapDriver.java

```diff
@@ -102,14 +109,19 @@ public String initialState() {
         log.trace("pollEvents: body={}", response.body());        
         JsonNode jsonNode = mapper.readTree(response.body());        
         final ArrayNode arrayNode = (ArrayNode) jsonNode;
+        Date maxTime = mapper.convertValue(arrayNode.get(0).get("receivedTimestamp"), Date.class);
```

Comment:
You need to add the startDate query parameter to the request URI on line 91. Otherwise, it requests all records every time, defeating the purpose of this PR.

Reply:
Done